packages/react-native-gesture-handler/src/v3/hooks/utils.ts

```diff
@@ -79,27 +88,91 @@ export function checkMappingForChangeProperties(animatedEvent: AnimatedEvent) {
   }
 }
 
-export function prepareConfig<THandlerData, TConfig>(
+export const commonGestureConfig = new Set<keyof CommonGestureConfig>([
+  'disableReanimated',
+  'enabled',
+  'shouldCancelWhenOutside',
+  'hitSlop',
+  'userSelect',
+  'activeCursor',
+  'mouseButton',
+  'enableContextMenu',
+  'touchAction',
+]);
+
+export const internalConfigProps = new Set<keyof InternalConfigProps<unknown>>([
+  'shouldUseReanimated',
+  'dispatchesAnimatedEvents',
+  'needsPointerData',
+  'changeEventCalculator',
+]);
+
+const PropsToFilter = new Set<BaseGestureConfig<unknown, unknown>>([
+  // Callbacks
+  'onBegin',
+  'onStart',
+  'onUpdate',
+  'onEnd',
+  'onFinalize',
+  'onTouchesDown',
+  'onTouchesMove',
+  'onTouchesUp',
+  'onTouchesCancelled',
+
+  // Config props
+  'changeEventCalculator',
+  'disableReanimated',
+
+  // Relations
+  'simultaneousWithExternalGesture',
+  'requireExternalGestureToFail',
+  'blocksExternalGesture',
+]);
+
+function withCommonProps<T extends string>(handlerProps: readonly T[]) {
+  return new Set<
+    T | keyof CommonGestureConfig | keyof InternalConfigProps<unknown>
+  >([...handlerProps, ...commonGestureConfig, ...internalConfigProps]);
+}
+
+export const PropsWhiteList = new Map<
+  SingleGestureName,
+  HandlersPropsWhiteList
+>([
+  [SingleGestureName.Pan, withCommonProps(panGestureHandlerProps)],
+  [SingleGestureName.Tap, withCommonProps(tapGestureHandlerProps)],
+  [SingleGestureName.Native, withCommonProps(nativeViewGestureHandlerProps)],
+  [SingleGestureName.LongPress, withCommonProps(longPressGestureHandlerProps)],
+  [SingleGestureName.Fling, withCommonProps(flingGestureHandlerProps)],
+]);
+
+const EMPTY_WHILE_LIST = new Set<string>();
```

Comment:
Similar thing will be added in #3689, so we will have to find good place to keep it 😅